### PR TITLE
add simple maintenance client

### PR DESCRIPTION
**Pull request by StupidHod**
I will add test sample with next commit.

### CODE REVIEW

File: src/main/java/com/coreos/jetcd/EtcdMaintenanceImpl.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    /**
+     * disarms a given alarm
+     *
+     * @param member the alarm
+     * @return the response result
+     */
+    @Override
+    public ListenableFuture<AlarmResponse> disAlarm(AlarmMember member) {
```
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/EtcdMaintenanceImpl.java]: where do we define AlarmMember?

File: src/main/java/com/coreos/jetcd/EtcdMaintenance.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+     * get the status of one member
+     */
+    ListenableFuture<StatusResponse> statusMember();
+
+    /**
+     * This handler will be called when the member make a snapshot
+     *
+     * @param snapshotRspStream
+     */
+    void setSnapshotObserver(StreamObserver<SnapshotResponse> snapshotRspStream);
```
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/EtcdMaintenance.java]: I do not quite understand how this method works. Do you have an example?

File: src/main/java/com/coreos/jetcd/EtcdMaintenanceImpl.java (hunk 1 above)
Comment (StupidHod) [inline src/main/java/com/coreos/jetcd/EtcdMaintenanceImpl.java]: AlarmMember is returned by listAlarms.

Comment (xiang90) [inline src/main/java/com/coreos/jetcd/EtcdMaintenanceImpl.java]: We probably should typedef it. We should not directly use the type from protobuf. I know we use them for return values today. But we will get rid of them soon.

File: src/main/java/com/coreos/jetcd/EtcdMaintenance.java (hunk 2 above)
Comment (StupidHod) [inline src/main/java/com/coreos/jetcd/EtcdMaintenance.java]: @xiang90  Ok, I will commit it with a example.

File: src/main/java/com/coreos/jetcd/EtcdMaintenanceImpl.java (hunk 1 above)
Comment (StupidHod) [inline src/main/java/com/coreos/jetcd/EtcdMaintenanceImpl.java]: @xiang90 Ok, I will convert it.

File: src/main/java/com/coreos/jetcd/EtcdMaintenance.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+     */
+    ListenableFuture<AlarmResponse> listAlarms();
+
+    /**
+     * disarms a given alarm
+     *
+     * @param member the alarm
+     * @return the response result
+     */
+    ListenableFuture<AlarmResponse> disAlarm(AlarmMember member);
```
Comment (hongchaodeng) [inline src/main/java/com/coreos/jetcd/EtcdMaintenance.java]: disalarm

**Comment by xiang90** [comment]
we need to get this rebased

**Comment by StupidHod** [comment]
@xiang90 OK, I have rebased this head to master.

File: src/main/java/com/coreos/jetcd/EtcdMaintenance.java (hunk 2 above)
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/EtcdMaintenance.java]: can we get an example for this? i hope we can get this pr merged soon :)

**Comment by StupidHod** [comment]
I have changed the interface for maintenance and add some tests for maintenance, but the alarm and snapshot is hard for me to trigger so the tests are not good. @xiang90

**Comment by hongchaodeng** [comment]
>  but the alarm and snapshot is hard for me to trigger so the tests are not good
Can we add a TODO for untested methods?

**Comment by StupidHod** [comment]
@hongchaodeng Good idea, I have done it with 'TODO comment' and detail how to use setSnapshotCallback in [testAddSnapshotCallback](https://github.com/coreos/jetcd/pull/38/files#diff-07b4e85b363c509c190262ba07d9c38cR49)

**Comment by hongchaodeng** [comment]
LGTM

**Pull request merged**